.github/workflows/linux_meson.yml

```diff
@@ -51,9 +51,22 @@ jobs:
         sudo apt-get install -y libopenblas-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache libmpc-dev
 
     - name: Install Python packages
+      if: matrix.python-version == '3.9'
       run: |
         python -m pip install numpy setuptools==59.8.0 wheel cython pytest pytest-xdist pytest-timeout pybind11 mpmath gmpy2 pythran ninja meson click rich-click doit pydevtool
 
+    - name: Install Python packages from repositories
+      if: matrix.python-version == '3.11-dev'
+      run: |
+        python -m pip install git+https://github.com/numpy/numpy.git
+        python -m pip install ninja cython pytest pybind11 pytest-xdist pytest-timeout click rich-click doit pydevtool
+        python -m pip install git+https://github.com/serge-sans-paille/pythran.git
+        python -m pip install git+https://github.com/mesonbuild/meson.git
+        python -m pip install -r mypy_requirements.txt
+
+        # Packages that are only needed for their annotations
+        python -m pip install types-psutil pybind11 sphinx
```

Comment:
Better to move these two lines into the `name: Mypy` block for clarity now I think.